### PR TITLE
Update dns managed zone test

### PR DESCRIPTION
**Pull request by slevenick**
This ignores the `dnssec_config` and `private_visibility_config` fields of dns managed zone. These could be added, but are not directly related to the test fix.

Fixes: https://github.com/terraform-providers/terraform-provider-google/issues/5036

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
`dns`: Added `visibility` field to `google_dns_managed_zone` data source
```

### CODE REVIEW

File: third_party/terraform/data_sources/data_source_dns_managed_zone.go [hunk 1]
```diff
@@ -27,15 +27,19 @@ func dataSourceDnsManagedZone() *schema.Resource {
 			},
 
 			"name_servers": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```
Comment (danawillow) [inline third_party/terraform/data_sources/data_source_dns_managed_zone.go]: Why this change?

Comment (slevenick) [reply third_party/terraform/data_sources/data_source_dns_managed_zone.go]: `name_servers` is a `TypeList` in the dns_managed_zone resource.

Between the time when this test last passed and now we changed from using a custom comparison method within this class to `checkDataSourceStateMatchesResourceState`. 

Comparison with the `checkDataSourceStateMatchesResourceState` method fails (I believe because of the different type) even though the list/set contain the same items. I believe because the new comparison compares on every attribute of the resource: https://github.com/terraform-providers/terraform-provider-google/blob/d0d6a895805f97bdf2a53197d325a2cf19f68c9b/google/test_utils.go#L110

while the old one only compared the whole `name_servers` object: https://github.com/terraform-providers/terraform-provider-google/blob/c605b0bc4ba2291d86b1038ce4b45b5ea50c1bb9/google/data_source_dns_managed_zone_test.go#L43
